sklearn/utils/estimator_checks.py

```diff
@@ -131,6 +131,29 @@ def _yield_classifier_checks(name, Classifier):
     if 'class_weight' in Classifier().get_params().keys():
         yield check_class_weight_classifiers
 
+def check_supervised_y_no_nan(name, Estimator):
+    # Checks that the Estimator targets are not NaN.
+
+    np.random.seed(888)
+    X = np.random.randn(10, 5)
```

Comment:
You should use a local random number generator here, as in 'rng = np.random.RandomState(888)' and not seed the global, as it create a side effect and is not concurrency safe.